bin/faucet/src/client.rs

```diff
@@ -79,7 +80,17 @@ impl FaucetClient {
         let mut rng = thread_rng();
         let coin_seed: [u64; 4] = rng.gen();
         let rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
-        Ok(Self { data_store, rpc_api, executor, id, rng })
+
+        let current_block_number = 0;
```

Comment:
You should be able to initialize this to the on-chain value using the `initialize_faucet_client` function